contrib/!source-control/github/packages.el

```diff
@@ -86,3 +88,36 @@
         "gfC" 'spacemacs/git-link-commit-copy-url-only)
       ;; default is to open the generated link
       (setq git-link-open-in-browser t))))
+
+;; magit-gh-pulls has to be loaded via a pre-init hook because the source code
+;; makes assumptions about the status of the magit-mode keymaps that are
+;; incompatible with the spacemacs' evilification feature.
+;; To avoid errors, magit-gh-pulls must be loaded after magit, but before magit
+;; is configured by spacemacs. 
+
+(defun github/init-magit-gh-pulls ())
+
+(defun github/pre-init-magit ()
+  (spacemacs|use-package-add-hook magit
+    :pre-config
+    (progn
+      (use-package magit-gh-pulls
+        ;; allow users to exclude magit-gh-pulls as they normally would
+        :if (configuration-layer/package-usedp 'magit-gh-pulls)
```

Comment:
@cmccloud you are right, this will bypass the exclusive property of init functions... mmhhhh maybe we could reverse the implementation, put it in `init-magit-gh-pulls` and check if magit is a used package.

Since your use-package hook is a `:pre-config` then it will be evaluated lazily so it is ok to define the hook even after `git/init-magit` has been evaluated.